core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
 
       @Override
-      synchronized CompressionCodec getCodec() {
-        if (codec == null) {
-          codec = new DefaultCodec();
-          codec.setConf(conf);
+      CompressionCodec getCodec() {
+        DefaultCodec resultCodec = codec.get();
+        if (null == resultCodec) {
+          DefaultCodec newCodec = new DefaultCodec();
+          newCodec.setConf(conf);
+          codec.compareAndSet(null, newCodec);
         }
 
-        return codec;
+        return codec.get();
```

Comment:
Seems like `return resultCodec` at the end would be more straightforward, no? Goes a bit to my other comment here.